os/fs/smartfs/smartfs_smart.c

```diff
@@ -277,10 +280,34 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 
 		if (parentdirsector != 0xFFFF) {
 			/* We can create in the given parent directory */
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+			if (smartfs_findinvalidentry(fs, parentdirsector, &invalidentry)) {
 
-			ret = smartfs_createentry(fs, parentdirsector, filename, SMARTFS_DIRENT_TYPE_FILE, mode, &sf->entry, 0xFFFF, sf);
-			if (ret != OK) {
-				goto errout_with_buffer;
+				invalidentry.firstsector = 0xFFFF;
+#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
```

Comment:
We will memset offset of invalid entry and put some info to there, am I right?
Then below is not necessary, IMOP, eh?

Reply:
The flags for the old invalidated entry will be different from what we need for the new entry.

So I think new flags need to be prepared so that they can be written to the entry.

Reply:
My comment was we'd better do this in replaceentry, but you want here?
Then Please add some comment. 
For example.
/* Now we found target invalidated entry. so update it before replaced */

Reply:
Understood, thanks